tests/test_categorical_generator.py

```diff
@@ -0,0 +1,24 @@
+import unittest
+import numpy as np
+from numpy.random import Generator, PCG64
+from synthetic_data.dataset_generators.categorical_generator import random_categorical
+
+
+class TestRandomsCategories(unittest.TestCase):
+    def setUp(self):
+        self.rng = Generator(PCG64(12345))
+        self.categories = ['X', 'Y', 'Z']
+
+    def test_default_return_validity(self):
+        result = random_categorical(self.rng, num_rows = 5)
```

Comment:
let's make sure to test at least two variation of num_rows to ensure functionality